### ISSUE TITLE
Problem with portals

### ISSUE THREAD

**Issue by wow2006** [GlPortal/glPortal#128]
- [x] Fire one portal make the player fall.
- [x]  Player change orientation after teleporting

**Comment by turlututututu** [comment]
Hello, just to be sure, are you referring to the fact that whatever your orientation be when you go through the portal you always end up as if you went through it head first?

Anyways, if nobody is on it, I would be glad to investigate one of those issues!

**Comment by hhirsch** [comment]
@turlututututu Yes. Thank you.

**Comment by turlututututu** [comment]
The interesting classes regarding this issue seem to be : `WorldHelper` and `PortalTeleport`. The issue is that orientation after teleporting is decided when the portal is constructed hence is independent of the player.

About the fall, I suspect action is default initialized to do nothing. So when hurting a wall, the teleportTrigger does nothing but since it is called it prevent the collision the be handled. Thus the player passes through the wall and fall.

**Comment by Geertiebear** [comment]
Fixed with PR #135

**Comment by turlututututu** [comment]
While I agree the "fall when only one portal exists" is fixed, the orientation thing is still present, at least on side.  So why closing this issue?

**Comment by hhirsch** [comment]
@turlututututu Can you clarify and say which steps to take to reproduce? The player exits the portal the way it is oriented, right?

**Comment by turlututututu** [comment]
@hhirsch Yeah the player exits the portal the way the portal is oriented, it is totally independent of  the player orientation! Which means you can go into a portal backward or sideways, it does not matter, you will always exit as if you went head first in the portal!

Beside the player orientation, the angle between the player and the portal is not preserved. Ie if you approach the portal diagonally you will exit the portal as if you went into it perfectly following the portal orientation.

In short, whatever the way you walk into the portal, whatever the direction you follow, the end result will be the same. And it seems counter-intuitive to me. I mean when I walk through a door the way I exit it depends of the way I entered it.

If I am unclear please let me know.

**Comment by Geertiebear** [comment]
I understand the issue now and I agree.

**Comment by hhirsch** [comment]
Fixed by #137
